.github/copilot-instructions.md

```diff
@@ -0,0 +1,137 @@
+# Overview
+This document provides guidelines for using GitHub Copilot to contribute to the .NET MAUI Community Toolkit. It includes instructions on setting up your environment, writing code, and following best practices specific to .NET MAUI.
+
+## Prerequisites
+1.	Install the latest stable (.NET SDK)[https://dotnet.microsoft.com/en-us/download].
+2.	Install .NET MAUI workloads (we recommend using Visual Studio installer).
+
+## Setting Up GitHub Copilot
+1.	Ensure you have GitHub Copilot installed and enabled in Visual Studio.
+2.	Familiarize yourself with the basic usage of GitHub Copilot by reviewing the (official documentation)[https://docs.github.com/en/copilot].
+
+## Writing Code with GitHub Copilot
+### General Guidelines
+* Use GitHub Copilot to assist with code completion, documentation, and generating boilerplate code.
+* Always review and test the code suggested by GitHub Copilot to ensure it meets the project's standards and requirements.
+
+### Specific to .NET MAUI
+* Ensure that any UI components or controls are compatible with .NET MAUI.
+* Avoid using Xamarin.Forms-specific code unless there is a direct .NET MAUI equivalent.
+* Follow the project's coding style and best practices as outlined in the (contributing)[https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md] document.
+
+## Best Practices
+* Use **Trace.WriteLine()** for debug logging instead of **Debug.WriteLine()**.
+* Include a **CancellationToken** as a parameter for methods returning **Task** or **ValueTask**.
+* Use **is** for null checking and type checking.
+* Use file-scoped namespaces to reduce code verbosity.
+* Avoid using the **!** null forgiving operator.
+** Follow naming conventions for enums and property names.
+
+### Debug Logging
+* Always use `Trace.WriteLine()` instead of `Debug.WriteLine` for debug logging because `Debug.WriteLine` is removed by the compiler in Release builds
+
+### Methods Returning Task and ValueTask
+* Always include a `CancellationToken` as a parameter to every method returning `Task` or `ValueTask`
+* If the method is public, provide a the default value for the `CancellationToken` (e.g. `CancellationToken token = default`)
+  * If the method is not public, do not provide a default value for the `CancellationToken`
+* Use `CancellationToken.ThrowIfCancellationRequested()` to verify the `CancellationToken`
+
+### Enums
+* Always use `Unknown` at index 0 for return types that may have a value that is not known
+* Always use `Default` at index 0 for option types that can use the system default option
+* Follow naming guidelines for tense... `SensorSpeed` not `SensorSpeeds`
+* Assign values (0,1,2,3) for all enums
```

Comment:
If they aren't marked with `Flags` attribute.

Not sure if the IA would be smart enough to know that.